src/preamble_minimal.js

```diff
@@ -99,6 +99,11 @@ if (!ENVIRONMENT_IS_PTHREAD) {
   updateGlobalBufferAndViews(wasmMemory.buffer);
 #if USE_PTHREADS
 } else {
+#if expectToReceiveOnModule('wasmMemory')
+  if (Module['wasmMemory']) {
+    wasmMemory = Module['wasmMemory'];
+  }
+#endif
```

Comment:
This will increase code size atm since `INCOMING_MODULE_JS_API` contains everything by default. I think that's not what we want in minimal runtime mode.

Perhaps as a first step we could in emcc.py make the default of `INCOMING_MODULE_JS_API` be the empty set in minimal runtime? Then in that mode all such extra code would need to be opted-in. Then this PR would not regress code size at all.

@juj does that sound like a reasonable path for minimal runtime?

Reply:
I think this should probably be unconditional, no?

```
#if MODULARIZE
wasmMemory = Module['wasmMemory'];
#endif
```

Then the line below wouldn't need the `MODULARIZE ?` thing?

Reply:
`#if MODULARIZE` would work I guess

Reply:
@sbc100 @kripken Let me know, happy to make this change

Reply:
I would like to see a test here.

And it that test should pass I think even if INCOMING_MODULE_JS_API is set to empty.   In other words, I won't think that expectToReceiveOnModule should appear here.   If the goal is to set wasmMemory then it should surely always be set correctly?

Reply:
I have to check why, but we can't do `wasmMemory = Module['wasmMemory'];` in non-modular path, it breaks on firefox. I guess it's just a matter of var orders or something I haven't looked deeper.

Reply:
Sure it can be guarded by `if MODULARIZE` maybe?

Maybe the best place to start would be simple test that fails today and is fixed by this change?